File: Semana_2/Playwright/Java/maven/curso/e2e/src/test/java/com/curso/playwright/pages/separated/LoginActions.java
package com.curso.playwright.pages.separated;

import com.curso.playwright.evidence.EvidenceSupport;
import com.microsoft.playwright.Page;

public class LoginActions {
    private final Page page;
    private final LoginLocators locators;

    public LoginActions(Page page) {
        this.page = page;
        this.locators = new LoginLocators(page);
    }

    public LoginActions open() {
        page.navigate(LoginLocators.URL);
        EvidenceSupport.capture(page, "parabank-separated-open-login");
        return this;
    }

    public void loginAs(String username, String password) {
        locators.usernameInput().fill(username);
        EvidenceSupport.capture(page, "parabank-separated-type-username");
        locators.passwordInput().fill(password);
        EvidenceSupport.capture(page, "parabank-separated-type-password");
        locators.loginButton().click();
        EvidenceSupport.capture(page, "parabank-separated-click-login");
    }
}

File: Semana_2/Playwright/TypeScript/npm/curso/e2e/pages/separated/login.actions.ts
import { Page } from '@playwright/test';
import { LOGIN_URL, LoginLocators } from './login.locators';
import { capture } from '../evidence';

export class LoginActions {
  private readonly locators: LoginLocators;

  constructor(private readonly page: Page) {
    this.locators = new LoginLocators(page);
  }

  async open(): Promise<LoginActions> {
    await this.page.goto(LOGIN_URL);
    await capture(this.page, 'parabank-separated-open-login');
    return this;
  }

  async loginAs(username: string, password: string): Promise<void> {
    await this.locators.usernameInput().fill(username);
    await capture(this.page, 'parabank-separated-type-username');
    await this.locators.passwordInput().fill(password);
    await capture(this.page, 'parabank-separated-type-password');
    await this.locators.loginButton().click();
    await capture(this.page, 'parabank-separated-click-login');
  }
}
